### PR TITLE
RELATED: ONE-3299 Fixed detection of overlapping labels on pie/donut

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1148/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/665/ : SUCCESS in 1m 40s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/368/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1153/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/178/ : UNSTABLE in 9m 26s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1159/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/673/ : SUCCESS in 2m 58s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/373/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1160/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/182/ : SUCCESS in 8m 54s

File: src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohidePieLabels.ts [hunk 1]
```diff
@@ -17,25 +18,29 @@ const autohidePieLabels = (chart: any) => {
...
         }
 
-        const intersects = isIntersecting(lastVisibleLabel, actualLabel);
-        if (!intersects) {
-            lastVisibleLabel = actualLabel;
-            actualLabel.show();
-        } else {
-            actualLabel.hide();
+        for (let neighborIdx = i + 1; neighborIdx < visiblePoints.length; neighborIdx++) {
+            const neighborLabel: any = get(visiblePoints, `${neighborIdx}.dataLabel`);
```
Comment (peterlacko) [inline src/components/visualizations/chart/highcharts/plugins/autohideLabels/autohidePieLabels.ts]: `const neighborLabel: any` -> `const neighborLabel: IRectBySize`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1174/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/683/ : SUCCESS in 4m 47s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1181/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/687/ : SUCCESS in 2m 41s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1183/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/690/ : SUCCESS in 5m 15s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/238/ : SUCCESS in 35s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/693/ : SUCCESS in 11m 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/389/ : SUCCESS in 13m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/135/ : SUCCESS in 13m 38s

**Pull request merged**